File: stack/stack.rb
class Stack
  def initialize
    @storage   = {}
    @size = 0
  end

  def push(value)
    @storage[@size] = value
    @size += 1
  end

  def pop
    if @size > 0
      temp = @storage[@size - 1]
      @storage.delete(@size -= 1);
      temp
    end
  end

  def size
    @size
  end

end

File: stack/stack.js
var Stack = function() {
  this.storage = {};
  this.size = 0;
}
Stack.prototype.push = function(value) {
  this.storage[this.size] = value;
  this.size += 1;
}
Stack.prototype.pop = function(){
  if(this.size) {
    var temp = this.storage[this.size - 1];
    delete this.storage[this.size--];
    return temp;
  }
}
Stack.prototype.length = function(){
  return this.size;
}
